pkg/hostpath/controllerserver.go

```diff
@@ -84,6 +90,23 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		glog.V(3).Infof("failed to create volume: %v", err)
 		return nil, err
 	}
+	if req.GetVolumeContentSource() != nil {
+		contentSource := req.GetVolumeContentSource()
+		if contentSource.GetSnapshot() != nil {
+			snapshotId := contentSource.GetSnapshot().GetId()
+			snapshot, ok := hostPathVolumeSnapshots[snapshotId]
+			if !ok {
+				return nil, status.Errorf(codes.NotFound, "cannot find snapshot %v", snapshotId)
+			}
+			snapshotPath := snapshot.Path
```

Comment:
Do we also check the snapshot status before using it? This won't matter here, but a good practice if this example is being picked for other places.

Reply:
Thanks for your review, updated